dotCMS/src/main/java/com/dotcms/rest/MapToContentletPopulator.java

```diff
@@ -316,6 +305,14 @@ private void processSeveralHostsOrFolders(final Contentlet contentlet,
 
                     Logger.warn(this, e.getMessage(), e);
                 }
+            } else if (query != null && query.trim().equals("")){
```

Comment:
I think the UtilMethods.isSet does this

Reply:
It is not exactly the same because UtilMethods.isSet also considers "null" string and null as a reference. And in this case, the logic has to be executed when query==""